docs/utils/singletons.md

```diff
@@ -0,0 +1,43 @@
+## Using and disposing singletons
```

Comment:
Think it would be good to (very briefly) also describe the use of singletons in Thunder (what they are) and as section on the code in Core  we have to help write singletons (e.g. SingletonType, the role of Core::Singleton::Dispose() etc.) and perhaps some small code example. If you have question on this I think Sebastion will be able to provide some details)